smarts/core/utils/logging.py

```diff
@@ -39,9 +39,13 @@ def timeit(name: str, log):
     """
     start = time()
     yield
-    elapsed_time = (time() - start) * 1000
+    elapsed_time = (time() - start) * 1000  # Units: milli seconds
 
-    log(f'"{name}" took: {elapsed_time:4f}ms')
+    if funcs:
+        for func in funcs:
+            func(elapsed_time)
```

Comment:
I'm a bit opposed to adding this to the `timeit` function. I think if you need complex behaviour like this, you should write a new function.